protocol/directory.go

```diff
@@ -321,3 +321,44 @@ func (d *ConiksDirectory) Monitor(req *MonitoringRequest) (
 
 	return NewMonitoringProof(aps, strs)
 }
+
+// GetSTRHistory gets the directory snapshots for the epoch range
+// indicated in the STRHistoryRequest req received from a CONIKS auditor.
+// The response (which also includes the error code) is supposed to
+// be sent back to the auditor. The returned error is used by the key
+// server for logging purposes.
+//
+// A request with a start epoch greater than the
+// latest epoch of this directory, or a start epoch greater than the
+// end epoch (if included) is considered malformed, and causes
+// GetSTRHistory() to return a
+// message.NewErrorResponse(ErrMalformedAuditorMessage) tuple.
+// GetSTRHistory() returns a message.NewSTRHistoryRange(strs) tuple.
+// strs is a list of STRs for
+// the epoch range [startEpoch, endEpoch], where startEpoch
+// and endEpoch are the epoch range endpoints indicated in the client's
+// request. If req.endEpoch is greater than d.LatestSTR().Epoch or
+// omitted in req, the end of the range will be set to d.LatestSTR().Epoch.
```

Comment:
We haven't checked if req.endEpoch is omitted in "MonitoringRequest", so should we change the range request to something like `[Epoch, Range]` where `Epoch` is `StartEpoch` and `Range` is the maximum number of epochs to return?